octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+        try:
+            if self.client.has.get('fetchMyTrades'):
+                return await self.exchange_manager.exchange.parse_trades(
+                    await self.client.fetchMyTrades(symbol=symbol, since=since, limit=limit, params=kwargs),
+                    check_completeness=check_completeness), None
+            else:
+                return None, f"Failed to fetch recent trades using get_my_recent_trades_default - " \
+                             "error: Exchange doesn't have a fetchMyTrades method\n"
+        except Exception as e:
+            return None, f"Failed to fetch recent trades using get_my_recent_trades_default - error: {e}"
+
+    async def get_my_recent_trades_using_recent_trades(self, symbol: str = None, limit: int = None,
+                                                       check_completeness: bool = True,
+                                                       **kwargs: dict) -> typing.Tuple[list or None, str or None]:
+        try:
+            return await self.get_recent_trades(symbol, limit, check_completeness, **kwargs), None
+        except Exception as e:
+            return None, f"Failed to fetch recent trades using get_my_recent_trades_using_recent_trades - error: {e}"
+
+    async def get_my_recent_trades_using_closed_orders(self, symbol: str = None, since: int = None,
+                                                       limit: int = None, check_completeness: bool = True,
+                                                       **kwargs: dict) -> typing.Tuple[list or None, str or None]:
+        try:
+            closed_orders = await self.get_closed_orders(symbol=symbol, since=since, limit=limit,
+                                                         check_completeness=check_completeness, **kwargs)
+            trades = []
+            for order in closed_orders:
+                if order[ecoc.STATUS.value] == OrderStatus.FILLED.value \
+                        or order[ecoc.STATUS.value] == OrderStatus.CLOSED.value:
+                    trades.append(order)
+            return trades, None
+        except Exception as e:
+            return None, f"Failed to fetch recent trades using get_my_recent_trades_using_closed_orders: {e}"
+
+    def cut_recent_trades_pagination_limit(self, limit: int) -> typing.Union[int, None]:
+        if self.connector_config.MAX_RECENT_TRADES_PAGINATION_LIMIT:
+            return min(self.connector_config.MAX_RECENT_TRADES_PAGINATION_LIMIT, limit)
         else:
-            raise octobot_trading.errors.NotSupported("This exchange doesn't support fetchMyTrades nor fetchTrades")
+            return limit
+
+    async def get_trade(self, trade_id, symbol, check_completeness=True) -> typing.Union[None, dict]:
+        trades = await self.get_my_recent_trades(symbol, check_completeness=check_completeness)
+        # usually the right trade is within the last ones
```

Comment:
Should we create a config entry for it?

Reply:
there is already one called GET_ORDER_METHODS.
get trade is only enabled in generic mode
Or do you mean a different config?